src/librustc_errors/styled_buffer.rs

```diff
@@ -111,7 +111,7 @@ impl StyledBuffer {
 
     pub fn prepend(&mut self, line: usize, string: &str, style: Style) {
         self.ensure_lines(line);
-        let string_len = string.len();
+        let string_len = string.chars().count();
```

Comment:
I would love to actually have a test that exercises this, but I recall having issues on some platforms with files with unicode chars in their paths. Could you test locally how this behave with compound characters, zero width characters and double width characters? In other parts of the codebase we use `UnicodeWidthChar::width(c).unwrap_or(0)` to properly align underlines, but in this case you're dealing with these as text buffers so what you did should be ok.